### DISCUSSION TITLE
Self-hosted runner: how to pass artifacts between jobs?

### DISCUSSION THREAD

**Discussion by divlv** [community/community#26165] [Actions]
<p>I’m using dockerized self-hosted runner ( <a href="https://github.com/myoung34/docker-github-actions-runner" rel="nofollow noopener">https://github.com/myoung34/docker-github-actions-runner</a> )</p>
<p>Looks like OK at the moment, but my question is:</p>
<p>how can I pass artifacts between jobs?<br>
I’ve tried <code>uses: actions/upload-artifact@v1</code> but it uploads an artifact directly to GitHub (with leads to free space problems)<br>
Can pass artifacts without uploading to GitHub?</p>

**Comment by rbellamy** [comment]
<p>Because the self-hosted runner isn’t ephemeral (as it is with the GitHub hosted runner), you can save artifacts in the local container for the next job to pick up. This will only work if you have a single self-hosted runner per repo - if you have more than one, then you’d have to find some external location to place the artifacts (e.g. copy them to the host, if the runners are all running in the same Docker or Kubernetes host, or push them to S3, etc.).</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/divlv">@divlv</a> ,</p>
<p>As doc metioned <a href="https://help.github.com/en/actions/configuring-and-managing-workflows/persisting-workflow-data-using-artifacts#about-workflow-artifacts" rel="nofollow noopener">here</a>, storing artifacts uses storage space on GitHub, you cannot store it in other places.</p>
<p>But if you’d like to persist data between different jobs on self-hosted runners, as <a class="mention" href="/rbellamy">@rbellamy</a> mentioned, you can use external location(eg: local file server…etc) to store data, access to it in the other job.</p>
<p>Thanks. </p>

**Comment by divlv** [comment]
<p>Is there any action like _actions/upload-artifact@v1 _available to store an artifact locally in an elegant way?..</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>I don’t find related actions, but since you’re using a self-hosted runner, you can simply zip the files and use ‘scp’ command to store the zip file into a file server. </p>
